### PR TITLE
Archive streams layout

### PR DESCRIPTION
**Pull request by JayTokBox**

### CODE REVIEW

**Comment by msach22** [comment]
@JayTokBox Could you please add an API proposal for this?

File: src/main/java/com/opentok/util/HttpClient.java [hunk 1]
```diff
@@ -380,7 +386,66 @@ public String setArchiveLayout(String archiveId, ArchiveProperties properties) t
                             " response code: " + response.getStatusCode());
             }
         } catch (InterruptedException | ExecutionException e) {
-            throw new RequestException("Could not delete an OpenTok Archive. archiveId = " + archiveId, e);
+            throw new RequestException("Could not set the layout, archiveId = " + archiveId, e);
+        }
+        return responseString;
+    }
+
+    public String setArchiveStreamLayout(String sessionId, StreamListProperties properties) throws OpenTokException {
```
Comment (jeffswartz) [inline src/main/java/com/opentok/util/HttpClient.java]: Please change this name to `setStreamsLayout`.

File: src/main/java/com/opentok/StreamProperties.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    public String id() {
+        return id;
+    }
+
+
+    /**
+     *   Returns layout class list for the stream.
+     *
+     */
+    public List<String>  getLayoutClass() {
```
Comment (jeffswartz) [inline src/main/java/com/opentok/StreamProperties.java]: Please change this name to `getLayoutClassList`.

File: src/main/java/com/opentok/util/HttpClient.java [hunk 3]
```diff
@@ -380,7 +386,66 @@ public String setArchiveLayout(String archiveId, ArchiveProperties properties) t
...
+
+            for(StreamProperties stream : properties.getStreamList()) {
+                jGenerator.writeStartObject();
+                jGenerator.writeFieldName("id");
+                jGenerator.writeString(stream.id());
+                jGenerator.writeFieldName("layoutClassList");
+                List<String> stringList = stream.getLayoutClass();
+                StringJoiner sj = new StringJoiner(",");
+                stringList.stream().forEach(e -> sj.add(doubleQuotes + e + doubleQuotes));
+                jGenerator.writeRawValue("["+ sj.toString() + "]");
```
Comment (jeffswartz) [inline src/main/java/com/opentok/util/HttpClient.java]: Will this work with an empty layout class list (to clear the classes for a stream)? In other words, if you do not call the `StreamProperties.Builder.addLayoutClass()` method when building the StreamProperties object, the stream's layout class list should be cleared.

Comment (JayTokBox) [reply src/main/java/com/opentok/util/HttpClient.java]: At present the code does
<code>
{"items":[
  {
    "id":"STREAMID1",
    "layoutClassList":[]
  }
  ]
}
</code>
and tested that it works

**Pull request merged**